packages/edit-site/src/components/navigation-sidebar/navigation-panel/menus/template-parts.js

```diff
@@ -21,14 +21,34 @@ import { MENU_ROOT, MENU_TEMPLATE_PARTS } from '../constants';
 
 export default function TemplatePartsMenu() {
 	const templateParts = useSelect( ( select ) => {
-		return select( 'core' ).getEntityRecords(
+		const publishedTemplateParts = select( 'core' ).getEntityRecords(
 			'postType',
 			'wp_template_part',
 			{
-				status: [ 'publish', 'auto-draft' ],
+				status: [ 'publish' ],
 				per_page: -1,
 			}
 		);
+		const currentTheme = select( 'core' ).getCurrentTheme()?.stylesheet;
+		const themeTemplateParts = select( 'core' ).getEntityRecords(
```

Comment:
alternatively, instead of doing 2 queries and combining the results we could do one query to retrieve all publish/auto-drafts and then filter the results to exclude auto drafts of other themes. I wonder which may be a better option?

Reply:
It seems having the 2 queries here both firing as the site editor starts loading sometimes messes up auto-draft creation and we get a duplicate of each template part supplied by the theme.  I would expect changes from #26383 to solve this since auto-drafts will have already been created by this point, but that still leaves the question above - combining 2 queries vs. filtering the results of 1 query.

Reply:
I don't have a strong opinion here, but the current two queries approach makes sense to me assuming that issue gets resolved

Reply:
Ah, I thought that PR was good to merge but it looks like you pointed out some other bugs with it.

I updated this PR to use the other approach to filter the results of one query instead, and it seems to be working well.